### PR TITLE
Update to ver. 2.1

### PR DESCRIPTION
**Pull request by dfandrich**
- Drop the filesystem capability now that the GUI uses a portal
- Update runtime to 23.08
- Update exiv2 to 0.28.2, with inih r58 as a new dependency
- Clean a few more files
- Use the upstream metainfo file

bot, build io.github.dfandrich.gpscorrelate

### CODE REVIEW

File: io.github.dfandrich.gpscorrelate.json [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
       "--socket=wayland",
       "--share=ipc",
-      /* Neeeded until the application uses a file chooser portal */
-      "--filesystem=host",
       /* Needed for gvfs to work */
       "--talk-name=org.gtk.vfs.*"
     ],
     "modules":[
+        {
+          "name": "inih",
```
Comment (hfiguiere) [inline io.github.dfandrich.gpscorrelate.json]: you can just disable INIH when building exiv2. It's only for the command line tool AFAIK.

File: io.github.dfandrich.gpscorrelate.json [hunk 2]
```diff
@@ -26,44 +39,45 @@
...
         },
         {
             "name": "gpscorrelate-gui",
             "buildsystem": "simple",
             "build-commands": [
-                /* gettext catalogues are dropped if stored in the standard location, so move them */
-                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\"",
+                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\" CXXFLAGS=\"${CXXFLAGS} -DENABLE_NLS=1\"",
                 "make prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale install install-po",
+                "sed -i 's/Icon=gpscorrelate-gui/Icon=io.github.dfandrich.gpscorrelate/' gpscorrelate.desktop",
```
Comment (hfiguiere) [inline io.github.dfandrich.gpscorrelate.json]: use `destkop-file-edit` instead.

File: io.github.dfandrich.gpscorrelate.json [hunk 3]
```diff
@@ -26,44 +39,45 @@
...
                 "cp gpscorrelate.desktop ${FLATPAK_ID}.desktop",
                 "desktop-file-install --vendor= --set-icon=${FLATPAK_ID} --dir=/app/share/applications ${FLATPAK_ID}.desktop",
                 "install -m0644 -Dp gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg",
+                "install -m0644 -Dp gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/gpscorrelate-gui.svg",
                 "install -m0644 -Dp ${FLATPAK_ID}.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo",
-                "make check"
+                "make check",
+                "install -m0755 -d ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps",
+                "gdk-pixbuf-thumbnailer -s 48 gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png",
+                "gdk-pixbuf-thumbnailer -s 48 gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/gpscorrelate-gui.png"
```
Comment (hfiguiere) [inline io.github.dfandrich.gpscorrelate.json]: you shouldn't need to do so since there is a SVG.

File: io.github.dfandrich.gpscorrelate.json (hunk 1 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json]: Good to know!

File: io.github.dfandrich.gpscorrelate.json (hunk 2 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json]: This just a hack to try to get the build to stop failing, but it made no difference.

File: io.github.dfandrich.gpscorrelate.json (hunk 3 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json]: This was a hack to figure out why the build is failing. Adding png icons didn't actually seem to make any difference.

Comment (hfiguiere) [reply io.github.dfandrich.gpscorrelate.json]:
```
  <launchable type="desktop-id">gpscorrelate.desktop</launchable>
``` 

In the appstream file, this is wrong.

A tip: there are top level properties for the manifest to do the rename.
`rename-appdata-file`, `rename-destkop-file` and `rename-icon`. They will take care of changing this in the right places. Which probably would remove many lines in the post-install.

Comment (hfiguiere) [reply io.github.dfandrich.gpscorrelate.json]: See the manpage for `flatpak-manifest` (or the online doc)

**Comment by dfandrich** [comment]
I can't figure out why the build/lint is failing. I've also posted about the problem at https://discourse.flathub.org/t/odd-appstream-errors-during-compose/6488/2

**Comment by hfiguiere** [comment]
the error now is in the appstream file, as it doesn't validate:

```xml
 <screenshots>
    <screenshot type="default">
      <image xml:lang="en">https://dfandrich.github.io/gpscorrelate/corr.png</image>
      <image xml:lang="fr">https://dfandrich.github.io/gpscorrelate/fr/corr.png</image>
    </screenshot>
  </screenshots>
```

https://www.freedesktop.org/software/appstream/docs/chap-Validation.html#asv-screenshot-image-no-source-but-en-locale

tl;dr, remove the `lang=en`

**Comment by dfandrich** [comment]
Bummer, I already fixed that one upstream but forgot to include it here.
